godot-git-plugin/src/git_api.cpp

```diff
@@ -203,39 +229,279 @@ Dictionary GitAPI::_get_modified_files_data() {
 
 			path = entry->head_to_index->new_file.path;
 		}
-		switch (entry->status) {
 
-			case GIT_STATUS_INDEX_NEW:
-			case GIT_STATUS_WT_NEW: {
+		static Dictionary map_changes;
+		map_changes[GIT_STATUS_WT_NEW] = 0;
+		map_changes[GIT_STATUS_INDEX_NEW] = 0;
+		map_changes[GIT_STATUS_WT_MODIFIED] = 1;
+		map_changes[GIT_STATUS_INDEX_MODIFIED] = 1;
+		map_changes[GIT_STATUS_WT_RENAMED] = 2;
+		map_changes[GIT_STATUS_INDEX_RENAMED] = 2;
+		map_changes[GIT_STATUS_WT_DELETED] = 3;
+		map_changes[GIT_STATUS_INDEX_DELETED] = 3;
+		map_changes[GIT_STATUS_WT_TYPECHANGE] = 4;
+		map_changes[GIT_STATUS_INDEX_TYPECHANGE] = 4;
+		map_changes[GIT_STATUS_CONFLICTED] = 5;
+
+		const static int GIT_STATUS_WT = GIT_STATUS_WT_NEW | GIT_STATUS_WT_MODIFIED | GIT_STATUS_WT_DELETED | GIT_STATUS_WT_TYPECHANGE | GIT_STATUS_WT_RENAMED | GIT_STATUS_CONFLICTED;
```

Comment:
You need not treat this variable as an enum. You can use normal camelCase for it